File: qmdc-ts/src/tokenizer.ts
/**
 * Markdown tokenizer wrapper using markdown-it
 */

import MarkdownIt from 'markdown-it';
import type Token from 'markdown-it/lib/token';

export function createTokenizer(): MarkdownIt {
  return new MarkdownIt({
    html: true,
  }).enable('table');
}

export function tokenize(markdown: string): Token[] {
  const md: MarkdownIt = createTokenizer();
  return md.parse(markdown, {});
}

File: qmdc-py/qmdc/tokenizer.py
"""Markdown tokenizer wrapper using markdown-it-py."""

from markdown_it import MarkdownIt
from markdown_it.token import Token


def create_tokenizer() -> MarkdownIt:
    """Create and configure markdown-it tokenizer."""
    md: MarkdownIt = MarkdownIt()
    md.enable("table")
    md.options["html"] = True
    return md


def tokenize(markdown: str) -> list[Token]:
    """Tokenize markdown content."""
    md: MarkdownIt = create_tokenizer()
    return md.parse(markdown)
